homework/index.js

```diff
@@ -16,31 +30,149 @@
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function loader(parent) {
+    const loaderDiv = createAndAppend('div', parent, { class: 'loaderDiv' });
+    createAndAppend('div', loaderDiv, { class: 'loader' });
+  }
+
+  function renderRepository(tableDiv, repositoryName) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}`;
+
+    fetchJSON(url, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      tableDiv.innerHTML = '';
+
+      const table = createAndAppend('table', tableDiv);
+      const tbody = createAndAppend('tbody', table);
+      const trForRepository = createAndAppend('tr', tbody, { class: 'flex-div' });
+      const trForDescription = createAndAppend('tr', tbody, { class: 'flex-div' });
+      const trForForks = createAndAppend('tr', tbody, { class: 'flex-div' });
+      const trForUpdated = createAndAppend('tr', tbody, { class: 'flex-div' });
+
+      createAndAppend('td', trForRepository, {
+        text: 'Repository:',
+        class: 'td-header',
+      });
+
+      const tdHyperlink = createAndAppend('td', trForRepository);
+      createAndAppend('a', tdHyperlink, {
+        text: `${data.name}`,
+        href: `${data.html_url}`,
+        target: '_blank',
+      });
+
+      createAndAppend('td', trForDescription, {
+        text: 'Description:',
+        class: 'description-td td-header',
+      });
+
+      createAndAppend('td', trForDescription, {
+        text: `${data.description}`,
+      });
+
+      createAndAppend('td', trForForks, {
+        text: 'Forks:',
+        class: 'td-header',
+      });
+
+      createAndAppend('td', trForForks, {
+        text: `${data.forks_count}`,
+      });
+
+      createAndAppend('td', trForUpdated, {
+        text: 'Updated:',
+        class: 'td-header',
+      });
+
+      createAndAppend('td', trForUpdated, {
+        text: `${data.updated_at}`,
+      });
+    });
+  }
+
+  function renderContributions(repositoryName, contributesDiv) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}/contributors`;
```

Comment:
Do not construct this url yourself. Use the `contributors_url` property from the repository object. GitHub may change the internal format of the url, but will not change the property name as that is part of the published API.